### PR TITLE
Made some changes

### PR DESCRIPTION
**Pull request by ujdcodr**
* Used docopt to improve command line argument parsing.
* Addressed the TODO issue of handling users that have already been verified.
Hopefully this works. docopt is some amazing stuff

### CODE REVIEW

**Comment by ujdcodr** [comment]
@chinmaydd  @therealkbhat  .... shall we begin the PR review process?

**Comment by ujdcodr** [comment]
I guess the main.py file is already pretty well documented. the cryptomagic going on in both main.py and client.py went over my head unfortunately. Pretty neat project I must say.

**Comment by ujdcodr** [comment]
There are some weird things about docopt though
* Upon typing ./client.py, it generates the dictionary with all the values of the corresponding option keys set to False(not harmful but still not good since it doesn't prompt the user about the correct Usage). I fixed this with the latest commit "Update client.py", i would call it a "jugaad" update.
* While giving incorrect options, only the correct usage is shown without showing the list of correct options(so the users have no idea of what are the correct options). It appears that this is also the case for commands like grep, however atleast the terminal leaves the message "Try grep --help for more information". When I tried placing that same text under the Usage section, the program went haywire so I just left it as it was.
* The wildest thing though, was that the dictionaries were returned even on typing commands like
  > ./client.py --configure
  > ./client.py --configur
  > ./client.py --configu
  > ./client.py --config
so and so forth

File: client.py [hunk 1]
```diff
@@ -105,21 +114,20 @@ def configure():
...
-    print originalMessage
+def main(docopt_args):
+    if docopt_args['--configure']:
+        configure()
+        print "Configuration successful!"
+    elif docopt_args['--encrypt']:
+        prePush(raw_input("Enter file path: "))
+    elif docopt_args['--decrypt']:
+        url = raw_input("Enter Dropbox share link: ")
+        originalMessage = postPull(url)
```
Comment (chinmaydd) [inline client.py]: Why do the two branches look different in code when the follow a similar pattern ?

File: main.py [hunk 2]
```diff
@@ -210,4 +225,4 @@ def decrypt():
     rv['dec_p'] = base64.b64encode(CaaS_decrypted_text)
     return rv
 
-run(host='localhost', port=8080)
+run(host='localhost', port=8080)s
```
Comment (chinmaydd) [inline main.py]: ?

File: main.py [hunk 3]
```diff
@@ -51,7 +52,17 @@ def verify_user(email_addr, secure_string):
...
+def already_verified(email_addr):
+    # Get value from the database
+    val = db.get(email_addr)
+    # Check if the email address in the database has already been verified
+    if val['confirmed'] == True:
+        return True
+    else:
+        return False
+
+# Function registers the user by adding data into the db and sending them a mail for authrntication via EBIA
```
Comment (chinmaydd) [inline main.py]: Add comments to functions following style as here mentioned [here](https://www.python.org/dev/peps/pep-0257/).

**Comment by chinmaydd** [comment]
@ujdcodr added some comments. Look into those.

**Comment by ujdcodr** [comment]
Ok

File: main.py (hunk 2 above)
Comment (ujdcodr) [inline main.py]: Random typo :-P

File: main.py (hunk 3 above)
Comment (ujdcodr) [inline main.py]: Should I rewrite comments for all functions based on PEP 257...or just for the new one(already_verified)?

File: client.py (hunk 1 above)
Comment (ujdcodr) [inline client.py]: Hence the indentation levels have been shifted by one....so i guess Git sees that as a modification.

Comment (chinmaydd) [inline client.py]: No, I am talking about the style:

```
Branch 1: prePush(raw_input("")
Branch 2: 
url = raw_input("")
postPull(url)
```

Comment (ujdcodr) [inline client.py]: I'm sorry, I didn't quite get you there....am i violating some style guide?

Comment (chinmaydd) [inline client.py]: Well, if you look at it both branches are calling similar functions but you have a variable initialization in one and direct argument passing in another. Doesnt that look a little strange? ;)

Comment (ujdcodr) [inline client.py]: oh ok

Comment (ujdcodr) [inline client.py]: fixed

Comment (chinmaydd) [inline client.py]: Better :+1:

**Comment by ujdcodr** [comment]
Anything else to take care of?

**Comment by chinmaydd** [comment]
Looks good for merge.

**Comment by chinmaydd** [comment]
@ujdcodr Thanks for the contribution. If you find any more bugs/features you would like to add, make sure to make a feature-branch next time and then raise a PR to master. That is the usual workflow in major repositories.

**Comment by ujdcodr** [comment]
Thank "you" @chinmaydd  and @therealkbhat for giving me this opportunity to learn. Got familiar with Python syntax and learned how to use docopt; couldn't ask for better mentors. Cheers.

**Comment by therealkbhat** [comment]
@ujdcodr thanks for taking the initiative.

Feel free to poke around both with this repo and others that we can assist you with. Have fun!

**Pull request merged**